src/test/java/net/hydromatic/lookml/test/LaxTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the LookML Authors under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.
+ * The LookML Authors license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.  You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ */
+package net.hydromatic.lookml.test;
+
+import net.hydromatic.lookml.LaxHandlers;
+import net.hydromatic.lookml.ObjectHandler;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.hasToString;
+
+/** Tests for the LookML event-based parser. */
+public class LaxTest {
+
+  private static void generateSampleEvents(ObjectHandler h) {
+    h.obj("model", "m", h1 ->
+            h1.number("n", 1)
+                .string("s", "hello")
+                .identifier("b", "true")
+                .code("code", "VALUES 1")
+                .list("list", h2 ->
+                    h2.identifier("asc")
+                        .number(-2.5)
+                        .string("abc")
+                        .list(h3 -> h3.string("singleton")))
+                .list("emptyList", h2 -> {}))
+        // This close() is spurious, but LookmlWriter currently needs it
+        // to trigger a flush of the internal state.
+        .close();
```

Comment:
This seems odd.  Why does LookmlWriter need `close` if `obj` already calls `close` on the handler?  I'm guessing it's because it's the `close` on the root obj?  If that's the case I suggest updating the comment to be something like:

```
        // This close() is the for the root object which must be manually called
```

The current comment makes me think that this `close` shouldn't be needed and the API is wrong.

Reply:
The `close()` method is necessary.

`obj` calls close on the sub-object's handler, but the root object is different, and still needs to be closed (even though it doesn't have a `}`).

I think I'll just remove the comment.

Reply:
I've improved the javadoc in commit 1136bf9.